File: frontend/src/types/controls.ts
export type ControlLevel = "bloquant" | "important" | "avertissement" | "information";

export interface ControlAnomaly {
  code: string;
  module: string;
  niveau: ControlLevel;
  titre: string;
  description: string;
  entreprise_id: string;
  exercice: number;
  objet_type: string;
  objet_id: string | null;
  route_frontend: string | null;
  metadata: Record<string, unknown>;
}

export interface ControlModuleSummary {
  module: string;
  statut: "ok" | "a_verifier" | "bloque";
  total_anomalies: number;
  bloquants: number;
  importants: number;
  avertissements: number;
  informations: number;
}

export interface PreClotureControls {
  entreprise_id: string;
  exercice: number;
  score: number;
  statut: "pret" | "a_verifier" | "bloque";
  resume: Record<string, ControlModuleSummary>;
  anomalies: ControlAnomaly[];
  avertissement_score: string;
  source_calcul: string;
}

File: backend/app/schemas/controls.py
"""Schemas API des controles globaux de pre-cloture V1."""
from __future__ import annotations

import uuid
from typing import Any, Literal

from pydantic import BaseModel, ConfigDict


NiveauControle = Literal["bloquant", "important", "avertissement", "information"]


class AnomalieControleOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    code: str
    module: str
    niveau: NiveauControle
    titre: str
    description: str
    entreprise_id: uuid.UUID
    exercice: int
    objet_type: str
    objet_id: uuid.UUID | None = None
    route_frontend: str | None = None
    metadata: dict[str, Any]


class ResumeModuleControleOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    module: str
    statut: str
    total_anomalies: int
    bloquants: int
    importants: int
    avertissements: int
    informations: int


class PreClotureOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    entreprise_id: uuid.UUID
    exercice: int
    score: int
    statut: str
    resume: dict[str, ResumeModuleControleOut]
    anomalies: list[AnomalieControleOut]
    avertissement_score: str
    source_calcul: str
